### DISCUSSION TITLE
Is it possible to use git config.credential.helper for signing commits, for example via ssh?

### DISCUSSION THREAD

**Discussion by ekhodx** [languitar/pass-git-helper#297] [Q&A] [answered]
I am looking for a way to "autofill" password which is being asked by git when signing commits with ssh key.

I know that there is `SSH_ASKPASS` env which can be used to a custom file which in theory can call _pass_ in order to get my password, but I can't really understand how to implement this script. I tried to log all incoming parameters - nothing.

Maybe I can implement script which will ask _pass_ for password depending on which ssh key we are trying to unlock, but I don't know how. Even when git asks you to sign password, it just displays generic password prompt with no path to ssh key or anything, in contrast of when I am using ssh auth key in order to push for example.

Thank you for this awesome tool!

UPD: This is a great demo of what I am talking about: https://youtu.be/BMFvhl0WRFQ How does git connects with 1Password in order to ask password for signing commit?

**Comment by languitar** [comment, marked as answer]
What is asking for a password there is SSH, not git. Generally, bridging that to pass is not in scope for pass-git-helper. Maybe you can hack something together with [sshpass](https://linux.die.net/man/1/sshpass)?

**Reply by ekhodx** [reply]
Thank you for this hint! I was researching a lot but still could not find something similar like sshpass. And yes, this is definitely out of pass-git-helper scope
